pkg/workflows/wasm/host/module.go

```diff
@@ -285,6 +300,15 @@ func NewModule(modCfg *ModuleConfig, binary []byte, opts ...func(*ModuleConfig))
 		return nil, fmt.Errorf("error wrapping fetch func: %w", err)
 	}
 
+	err = linker.FuncWrap(
+		"env",
+		"emit",
+		createEmitFn(logger, modCfg.Emitter),
```

Comment:
This should be context-aware, by leveraging the requestId + the store to add state to the ongoing request. Also we need to add a set of predefined tags (executionID, owner, workflowID, name -- see the ticket for the list) to each emission.